ngraph/core/include/ngraph/op/util/evaluation_context.hpp

```diff
@@ -0,0 +1,109 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include "ngraph/op/util/variable.hpp"
+#include "ngraph/op/util/variable_value.hpp"
+
+namespace ngraph
+{
+    class Variable;
+    class VariableValue;
+
+    using VariablePtr = std::shared_ptr<Variable>;
+    using VariableValuePtr = std::shared_ptr<VariableValue>;
+    using VariableMap = std::unordered_map<VariablePtr, VariableValuePtr>;
+
+    /// VariableContext stores and manages a evaluation context for Variables.
+    class VariableContext
+    {
+    public:
+        /// \brief Constructs an uninitialized VariableContext.
+        VariableContext() = default;
+
+        /// \brief Constructor for VariableContext.
+        /// \param variable_values The values associated with a particular Variables.
+        VariableContext(const VariableMap& variable_values)
+            : m_variable_values(variable_values)
+        {
+        }
+
+        /// \brief Sets the reset flags for all stored Variables to true.
+        void reset_variable_context()
+        {
+            for (const auto& el : m_variable_values)
+            {
+                el.second->set_reset(true);
+            }
+        }
+
+        /// \brief Sets the new values for Variables.
+        /// \param variable_values The new values associated with a particular Variable.
+        void set_variable_values(const VariableMap& variable_values)
+        {
+            m_variable_values = variable_values;
+        }
+
+        /// \brief Changes/sets the values for Variable.
+        /// \param variable New or stored Variable.
+        /// \param variable_value The values associated with the variable.
+        void add_variable_value(const VariablePtr& variable, const VariableValuePtr& variable_value)
```

Comment:
add_variable_value looks like adding value in array for me, possibly use set_variable_value is better

Reply:
done